.github/workflows/soakTest.yml

```diff
@@ -26,124 +30,164 @@ jobs:
         run: |
           npm install -g yarn
           yarn install
           yarn setup
+
+      - name: Install SOPS
+        run: |
+          TEMP_DEB="temp-sops_3.7.1_amd64.deb" &&
+          wget -O "$TEMP_DEB" https://github.com/mozilla/sops/releases/download/v3.7.1/sops_3.7.1_amd64.deb &&
+          sudo dpkg -i "$TEMP_DEB" &&
+          rm -f "$TEMP_DEB"
+
+      - name: Install Helm Secrets
+        run: helm plugin install https://github.com/jkroepke/helm-secrets --version v3.12.0
+
       - name: Use GH CLI to get pr info
         id: get-pr-info
         env:
           GITHUB_TOKEN: ${{ secrets.QA_GITHUB_TOKEN }}
           PR_NUMBER: ${{ steps.findPr.outputs.pr }}
         run: |
-          set -x
           CHANGED_FILES_NAME=./changedFiles.txt
           gh pr view ${PR_NUMBER} --json files --jq '.files.[].path' > ${CHANGED_FILES_NAME}
-          ADAPTERS_LIST="$(yarn get-changed-adapters "${CHANGED_FILES_NAME}")"
+          TEST_ADAPTERS="$(yarn get-changed-adapters "${CHANGED_FILES_NAME}")"
+
+          echo "::set-output name=TEST_ADAPTERS::${TEST_ADAPTERS}"
 
-          # export adapters from this step
-          echo "::set-output name=ADAPTERS_LIST::${ADAPTERS_LIST}"
-      - name: Remove adapters that have no fluxconfig
-        id: adapters-list
+      - name: Clone adapter-secrets repo
         env:
-          ADAPTER_NAMES: ${{ steps.get-pr-info.outputs.ADAPTERS_LIST }}
+          GITHUB_TOKEN: ${{ secrets.QA_GITHUB_TOKEN }}
+        run: git clone https://${GITHUB_TOKEN}@github.com/smartcontractkit/adapter-secrets.git
+
+      - name: Build the k6 payloads and images
+        id: k6-payloads
+        env:
+          PR_NUMBER: ${{ steps.findPr.outputs.pr }}
+          TEST_ADAPTERS: ${{ steps.get-pr-info.outputs.TEST_ADAPTERS }}
+          IMAGE_PREFIX: ${{ secrets.QA_AWS_ACCOUNT_NUMBER }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/k6
         run: |
-          set -x
-          ADAPTERS_LIST=
-          for adapter in ${ADAPTER_NAMES}; do
-            echo "${adapter}"
-            DONT_ADD=0
-            yarn qa:flux:configure exists ${adapter} blank || DONT_ADD=1
-            if [ $DONT_ADD = '1' ]; then
-              continue
-            fi
-            ADAPTERS_LIST="${ADAPTERS_LIST} ${adapter}"
-          done
-          ADAPTERS_LIST="$(echo ${ADAPTERS_LIST} | xargs)"
-          # export adapters from this step
-          echo "::set-output name=ADAPTERS_LIST::${ADAPTERS_LIST}"
+          if [ -n "$TEST_ADAPTERS" ]; then
+            for adapter in ${TEST_ADAPTERS}; do
+              yarn qa:flux:configure k6payload ${adapter} ${PR_NUMBER} || exit 1
+
+              cd ./packages/k6
+
+              echo "" >> ./.env
+              echo "PAYLOAD_GENERATED=true" >> ./.env
+              echo "CI_ADAPTER_NAME=${adapter}" >> ./.env
+              echo "QA_RELEASE_TAG=${PR_NUMBER}" >> ./.env
+              echo "PR_NUMBER=${PR_NUMBER}" >> ./.env
+              echo "TEST_DURATION=10m" >> ./.env
```

Comment:
Just flagging to make sure, `TEST_DURATION=10m` is good?

Reply:
We can re-adjust later on if we see that's necessary